client/src/pages/Settings/Availability/DropDownItem.tsx

```diff
@@ -0,0 +1,723 @@
+import { Formik, Field, Form, useField } from 'formik';
+import { Select, Checkbox, Typography, Table, TableBody, TableCell, TableContainer, TableRow } from '@mui/material/';
+
+import React from 'react';
+
+const DropDownItem: React.FC = () => {
+  const schedulesType = [{ name: 'work week' }, { name: 'holiday week' }];
+  const options: { value: string; label: string }[] = [];
+  for (let i = 0; i < 25; i++) {
+    i < 10
+      ? options.push({ value: `0${i}:00`, label: `0${i}:00` })
+      : options.push({ value: `${i}:00`, label: `${i}:00` });
+  }
+  return (
+    <div>
+      <Typography align="center" variant="h5" component="div" fontWeight={600} marginBottom={7} gutterBottom>
+        Your availability
+      </Typography>
+      <Formik
+        initialValues={{
+          schedules: new Map<string, { [key: string]: any }>(),
+          scheduleType: '',
+          days: [],
+        }}
```

Comment:
Just playing around with it locally and I think this is overcomplicated. A simpler approach would be just to for every day of the week render the row and just track whether it's been activated or not via the checkbox. So, in the map you would pass down the values, setFieldValue, etc and then render the form. This will significantly reduce the TSX in this file and make it more concise, I hope this makes sense.

Reply:
The structure of incoming data to be expected will be:

```
{
   monday: {
      active: boolean,
       startTime: Number,
       endTime: Number
   },
   . . .
}
```

Reply:
Albeit, probably not the most efficient solution but I think it's a more simpler approach and I'm okay to go down this approach 👍🏻

Reply:
but don't you want to separate the different schedules (and in the future add new ones )
the sitter might have 2 different schedules that he toggles between

Reply:
so some thing like this  
and formik automatically takes out the disabled fields 
`{
    schedules : {
         workWeek :{
              days :{
                  sunday: {
                     startTime: 
                      endTime           
                    }
              }
         }
  }
}`

Reply:
Yeah - so the complexity for this might be a bit but essentially for viewing availabilities we want to set the initial values to the selected availability then when clicking "New Schedule" we want to clear the initial values and send down the name to the post handler and when submitting an already created form we just want to update it.